zulipterminal/model.py

```diff
@@ -1247,6 +1254,29 @@ def make_reduced_stream_data(stream: Subscription) -> StreamData:
             new_visual_notified_streams
         )
 
+    def _unsubscribe_from_streams(
+        self, subscriptions: List[RemovedSubscription]
+    ) -> None:
+        for subscription in subscriptions:
+            stream_id = subscription["stream_id"]
+            self.pinned_streams[:] = [
+                stream
+                for stream in self.pinned_streams
+                if stream.get("id") != stream_id
+            ]
+            self.unpinned_streams[:] = [
+                stream
+                for stream in self.unpinned_streams
+                if stream.get("id") != stream_id
+            ]
+            if stream_id in self.muted_streams:
+                self.muted_streams.remove(stream_id)
+            if stream_id in self.visual_notified_streams:
+                self.visual_notified_streams.remove(stream_id)
+            if stream_id in self.stream_dict:
+                self.unsubscribed_stream_dict[stream_id] = self.stream_dict[stream_id]
+                del self.stream_dict[stream_id]
```

Comment:
Readability: for comparison with the subscription function, this should maybe go at the top of the loop?